### DISCUSSION TITLE
Webdav

### DISCUSSION THREAD

**Discussion by yeah-mike** [ggogel/seafile-containerized#11] [Q&A] [answered]
Does this implementation support webdav? I'm having difficulty determining which docker should be exposing the webdav port

**Comment by ggogel** [comment, marked as answer]
Yes, webdav is supported. The seafile-server service handles webdav. The port is already exposed and the path `/seafdav` is redirected by seafile-caddy to the port `8080` on seafile-server.

Create the following config file: `seafile-data/seafile/conf/seafdav.conf`
```
[WEBDAV]
enabled = true
port = 8080
fastcgi = false
share_name = /seafdav
```

Then restart seafile-server.

**Reply by ggogel** [reply]
Also make sure to use `ggogel/seafile-caddy:1.0.3`, which contains a fix regarding seafdav.

**Reply by yeah-mike** [reply]
Thanks. Using seafile-caddy:1.0.3 fixed the issue with the trailing /.

I do seem to have another issue with webdav. It seems creating, editing, and deleting files work fine, but any time moving is involved it fails. For example, if I wanted to rename the file "test1.txt" as "test2.txt" within the library "Vault", here is the error I see in logs for seafile-server:

`2021-05-18 14:34:09.035 - <23392548173568> wsgidav.wsgidav_app INFO : 172.21.0.2 - username@gmail.com - [2021-05-18 21:34:09] "MOVE /Vault/test1.txt" dest="https://seafile.myserver.com/seafdav/Vault/test22.txt", depth=0, overwrite=F, elap=0.019sec -> 502 Bad Gateway`

I suspect it has something to do with the caddy, because if I expose/export port 8080 on seafile-server and connect directly with it the operation succeeds.

Sorry if I'm not providing enough background information. This is my first time using docker so I'm quite new to it. Let me know if I can include any other log files/config files.

**Reply by yeah-mike** [reply]
On further research, this seems to only be a problem if https is used. The "fix" appears to be described in this thread: https://forum.seafile.com/t/seafdav-move-command-causing-502/11582

**Reply by ggogel** [reply]
I just found the same thread. I'm trying to implement a fix right now.

**Reply by ggogel** [reply]
I think I've found the corresponding setting in caddy. Though, I'm not sure how to reproduce this issue. Can you test if the fix in `ggogel/seafile-caddy:dev` works for you?

**Reply by yeah-mike** [reply]
Thanks @ggogel, `ggogel/seafile-caddy:dev` appears to have mostly fixed it.

I say "mostly" because everything works on 3 out of 4 webdav clients I tried. The only webdav client that doesn't work completely is Cx File Explorer ( https://play.google.com/store/apps/details?id=com.cxinventor.file.explorer ). With Cx File Explorer, uploading a new file or editing an existing file will fail with a network error. Again, everything works as intended if I connect directly with seafile-server on port `8080`. Unfortunately the logs don't really show anything useful, as curiously there isn't even a `PUT` command written to the log. However, since the 3 other webdav clients work fine, I'm inclined to believe it's mostly the fault of Cx as opposed to Seafile.
